Pages/CodeButton.js

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useContext } from "react";
+import {
+  Text,
+  View,
+  StyleSheet,
+  TextInput,
+  Button,
+  Alert,
+  Modal,
+  TouchableHighlight,
+} from "react-native";
+import { useMutation, gql } from "@apollo/client";
+import { useForm, getErrors } from "../util/hooks";
+import Constants from "expo-constants";
+
+function CodeButton(props) {
+  const [errors, setErrors] = useState({});
+  const [modalVisible, setModalVisible] = useState(false);
+
+  let user = props.user;
+
+  const { values } = useForm(redeemPointsCallback, {
+    code: "",
+    username: user.username,
+  });
+
+  function redeemPointsCallback() {
+    redeemPoints();
+  }
+
+  const [redeemPoints] = useMutation(REDEEM_POINTS_MUTATION, {
+    update(_, { data: { redeemPoints: userData } }) {
+      values.code = "";
+      setErrors(false);
+      setModalVisible(false);
+      updateGetUser(userData);
+    },
+
+    onError(err) {
+      getErrors(err);
+    },
+
+    variables: values,
+  });
+
+  function updateGetUser(userData) {
```

Comment:
get rid of all this stuff for updating the user, this won't effect any change outside of the scope of this component (so nothing). instead, there's a way to update the local apollo cache which we you don't have to worry about for now (examples in the shpe uf client). Once we stitch everything together we will start adding in proper refreshing and updating after mutations